### PR TITLE
URI completion

### PR DESCRIPTION
**Pull request by maximilianfridrich**
Adds function `uri_complete` to uri.c.

Relates to:

- https://github.com/baresip/baresip/pull/2135

### CODE REVIEW

File: src/uri/uri.c [hunk 1]
```diff
@@ -271,3 +273,75 @@ int uri_headers_apply(const struct pl *pl, uri_apply_h *ah, void *arg)
...
+		!sa_set_str(&sa_addr, host, 0);
+	if (!uri_is_ip && host[0] == '[' && (c = strchr(host, ']'))) {
+		*c = 0;
+		uri_is_ip = !sa_set_str(&sa_addr, host+1, 0);
+	}
+
+	mem_deref(uridup);
+
+	if (0 != re_regex(uri, len, "[^@]+@[^]+", NULL, NULL) &&
+		1 != uri_is_ip && hostport != NULL) {
```
Comment (cspiel1) [inline src/uri/uri.c:341]: Why `1 != uri_is_ip`? It is a bool.

Comment (maximilianfridrich) [reply src/uri/uri.c]: Thank you. Fixed.

File: src/uri/uri.c [hunk 2]
```diff
@@ -271,3 +273,75 @@ int uri_headers_apply(const struct pl *pl, uri_apply_h *ah, void *arg)
...
+/**
+ * Auto complete a SIP uri, add scheme and domain if missing
+ *
+ * @param uri Input SIP uri
+ * @param hostport host[:port] of SIP URI
+ * @param buf Target buffer to print SIP uri
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int uri_complete(const char *uri, const char *hostport, struct mbuf *buf)
```
Comment (alfredh) [inline src/uri/uri.c:287]: please note that `struct uri` is generic URI code

this new function is specific to SIP only.

I think this function is too specific to belong in a general purpose library.
It is better to be located in baresip, which is also a library -> libbaresip.so

**Comment by maximilianfridrich** [comment]
Thank you for the review. I will close this PR as a result of the disccussion in baresip/baresip#2135 and your comments.

**Pull request closed without merging**